oppia/oppia-android
# Fixes part of #40 & #42: Generalisation Highfi Mobile Portrait + Landscape - Buttons

## Explanation


This PR places the buttons inside state-recycler-view in correct positions. Because of the inconsistency in the mocks the the margin values are not perfect. But to make sure the UI looks great following assumptions have been made.
Also, this PR fixes the DragAndDrop internal item positioning too.

1. The left side button should follow the outer-most margin, meaning it should align with the content-item on left side.
2. The right side button should also follow the outer-most margin, meaning it should align with the interaction-item on right side.
3. Margin top for all items is 24dp in mobile and 72dp in tablet. 
4. Margin bottom is 0dp
5. Buttons also contain a `4dp` margin so that we can show elevation-shadow correctly and as a result the parent layout margins are `4dp` less than that mentioned in the mocks. This solution ultimately result in UI similar to the mocks.
6. `Congratulations` text follow the `match_parent` width and makes sure that it follows out most margins on both sides.

## How to test
1. Before you start testing, make sure that in each screen you first enter 2 wrong answers so that you can easily see if the items are correctly aligned or not.
2. Run `Prototype Exploration` in **mobile** in **landscape** as well as **portrait** mode.
3. Run `Prototype Exploration` in **tablet** in **landscape** as well as **portrait** mode.
4. Run `Fractions Question Player` in **mobile** in **landscape** as well as **portrait** mode.
5. Run `Fractions Question Player` in **tablet** in **landscape** as well as **portrait** mode.

## Reference Mocks
https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/screen/216ef636-187a-4c7a-b420-1aa48e95586d/
https://xd.adobe.com/view/d405de00-a871-4f0f-73a0-f8acef30349b-a234/screen/4ba7d54d-9b20-47cd-89c3-2d69d9001981/
https://xd.adobe.com/view/d405de00-a871-4f0f-73a0-f8acef30349b-a234/screen/f136ee72-dd2e-42cb-8ff3-7b1cb6d4eb04/
https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/screen/7491be91-a457-4f24-af0b-a9df902bee39/
https://xd.adobe.com/view/d405de00-a871-4f0f-73a0-f8acef30349b-a234/screen/c6c94f34-f3a1-43ce-9ed1-7b9a7d6dd81e/
https://xd.adobe.com/view/d405de00-a871-4f0f-73a0-f8acef30349b-a234/screen/19b5ba0c-346c-4fd5-9699-f270c0434b6c/
https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/grid

https://xd.adobe.com/view/d405de00-a871-4f0f-73a0-f8acef30349b-a234/screen/4d120159-d03f-4f48-82c0-f162565e5745/

## Checklist
